### PR TITLE
PHPCS: The remaining endpoint files

### LINKED ISSUE #21957: PHPCS: JSON Endpoints M-T
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-menus-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-render-embed-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-render-embed-reversal-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-render-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-render-shortcode-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-sharing-buttons-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-v1-3-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-v1-4-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-user-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-taxonomy-endpoint.php",
```

### PR DESCRIPTION
**Pull request by sdixon194**
Fixes #21957

#### Changes proposed in this Pull Request:

* PHPCS Changes for the rest of the endpoint files

#### Jetpack product discussion

N/a
#### Does this pull request change what data or activity we track or use?

No
#### Testing instructions:

* Proof and test?

### CODE REVIEW

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-user-endpoint.php [hunk 1]
```diff
@@ -189,24 +224,25 @@ public function update_user( $user_id, $blog_id ) {
...
+			} elseif ( is_string( $input['roles'] ) ) {
 				$user['role'] = $input['roles'];
 			} else {
 				return new WP_Error( 'invalid_input', __( 'The roles property must be a string or an array.', 'jetpack' ), 400 );
 			}
 
 			$editable_roles = array_keys( get_editable_roles() );
-			if ( ! in_array( $user['role'], $editable_roles ) ) {
+			if ( ! in_array( $user['role'], $editable_roles, true ) ) {
+				// translators: The role that was returned that's not valid.
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-user-endpoint.php:245]:
```suggestion
				// translators: The role that is not valid.
```

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-endpoint.php [hunk 2]
```diff
@@ -516,21 +563,25 @@ public static function find_featured_media( &$post ) {
...
+	 * @param array $attr - the attribute.
+	 */
+	public function win8_gallery_shortcode( $attr ) {
 		global $post;
 
 		static $instance = 0;
 		$instance++;
 
-		$output = '';
+		$output = ''; // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-endpoint.php:578]: > r if no one cares about Windows 8 anymore anyway and we can lose this whole thing.

This is what I was hoping for :D But it scared me how it does _seem_ like something that is intended to be returned, but isn't. In any case, I doubt it's used much at all anyway.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-v1-1-endpoint.php [hunk 3]
```diff
@@ -324,19 +368,25 @@ function get_blog_post( $blog_id, $post_id, $context = 'display' ) {
...
+	 * @param array $attr - the attribute.
+	 */
+	public function win8_gallery_shortcode( $attr ) {
 		global $post;
 
 		static $instance = 0;
 		$instance++;
 
-		$output = '';
+		$output = ''; // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-v1-1-endpoint.php:383]: Went ahead and added to-dos

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-user-endpoint.php [hunk 4]
```diff
@@ -148,31 +172,42 @@ function callback( $path = '', $blog_id = 0, $user_id = 0 ) {
...
 	public function update_user( $user_id, $blog_id ) {
-		$input = $this->input();
+		$user       = array();
+		$input      = $this->input();
 		$user['ID'] = $user_id;
-		$is_wpcom = defined( 'IS_WPCOM' ) && IS_WPCOM;
+		$is_wpcom   = defined( 'IS_WPCOM' ) && IS_WPCOM;
 
-		if ( get_current_user_id() == $user_id && isset( $input['roles'] ) ) {
+		if ( get_current_user_id() === $user_id && isset( $input['roles'] ) ) {
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-user-endpoint.php:206]: Good idea.

**Review by jeherve — COMMENTED** [review]
This looks good. I just had two minor notes.

**Comment by sdixon194** [comment]
Should be good for a final review. Also had to add some wpcom related unslashing and sanitization, such as [here](https://github.com/Automattic/jetpack/blob/phpcs/json-endpoints-m-t/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-v1-1-endpoint.php#L108) and [here](https://github.com/Automattic/jetpack/blob/phpcs/json-endpoints-m-t/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-post-v1-1-endpoint.php#L326). Let me know if those seem okay.

**Review by jeherve — APPROVED** [review]
Nice work on all of those! 🚢

**Comment by sdixon194** [comment]
r244778-wpcom

**Pull request merged**